### PR TITLE
Re-enable windows for nym-vpn-core CI and release

### PR DESCRIPTION
**Pull request by octol**
This change is [Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/1401)

### CODE REVIEW

**Review by rokas-ambrazevicius — APPROVED** [review]

**Review by pronebird — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/1401#-:-OAHRcaQ-J5xZv6855Vf:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @octol)

**Review by pronebird — CHANGES_REQUESTED** [review]
Working on introducing msys2 that I know use to build wireguard-go

**Comment by pronebird** [comment]
Main build is all green: https://github.com/nymtech/nym-vpn-client/actions/runs/11568586122/job/32201088230

**Review by pronebird — APPROVED** [review]

File: .github/workflows/ci-nym-vpn-core-windows.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        with:
+          toolchain: stable
+          components: rustfmt, clippy
+
+      - name: Set env
+        shell: bash
+        run: |
+          triplet=x86_64-pc-windows-msvc
+          echo "TRIPLET=$triplet" >> $GITHUB_ENV
+          echo "RUSTFLAGS=-L ${GITHUB_WORKSPACE}/build/lib -Clink-args=/LIBPATH:${GITHUB_WORKSPACE}/build/lib/x64-Debug" >> $GITHUB_ENV
```
Comment (octol) [inline .github/workflows/ci-nym-vpn-core-windows.yml:45]: I thought we didn't need this anymore @pronebird ?

File: .github/workflows/ci-nym-vpn-core-windows.yml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        env:
+          MSYS2_PATH_TYPE: inherit
+        run: |
+          ./wireguard/build-wireguard-go.sh
+
+      - name: Download wintun.zip and winpcap.zip
+        shell: bash
+        run: |
+          curl --output ${GITHUB_WORKSPACE}/wintun.zip https://www.wintun.net/builds/wintun-0.14.1.zip
+          curl --output ${GITHUB_WORKSPACE}/winpcap.zip https://www.winpcap.org/install/bin/WpdPack_4_1_2.zip
```
Comment (octol) [inline .github/workflows/ci-nym-vpn-core-windows.yml:100]: I think this one is unused, let me try removing it

File: .github/workflows/ci-nym-vpn-core-windows.yml (hunk 1 above)
Comment (octol) [reply .github/workflows/ci-nym-vpn-core-windows.yml:45]: oh I see now, it's for winfw et al

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nymtech/nym-vpn-client/1401)* status: 1 of 7 files reviewed, 1 unresolved discussion

___
*[`.github/workflows/ci-nym-vpn-core-windows.yml` line 45 at r8](https://reviewable.io/reviews/nymtech/nym-vpn-client/1401#gh-1820232534:-OAMLMsnC_zo1TEI3f9t:b-jpw8kf) ([raw file](https://github.com/nymtech/nym-vpn-client/blob/d2ee1f40afca0c1f29c210e66cb21f823c0c6de5/.github/workflows/ci-nym-vpn-core-windows.yml#L45)):*
<details><summary><i>Previously, octol (Jon Häggblad) wrote…</i></summary><blockquote>

oh I see now, it's for winfw et al
</blockquote></details>

Yes. Some of the firewall libs are put in there. We don't really use firewall but I didn't want to remove these steps because soon we'll need them again.

**Pull request merged**